azure-mgmt-resources/src/main/java/com/azure/management/RestClient.java

```diff
@@ -8,31 +11,40 @@
 
 public class RestClient {
 
-    private final RestAsyncClient client;
+    private HttpPipeline pipeline;
+
+    private final URL baseUrl;
+
+    /**
+     * The original builder for this rest client.
+     */
+    private final RestClientBuilder builder;
 
-    public RestClient(RestAsyncClient client) {
-        this.client = client;
+    public RestClient(URL baseUrl, HttpPipeline pipeline, RestClientBuilder builder) {

```

Comment:
Not usual to take a Builder as parameter for constructor. And if we go from Builder, we probably do not need a public constructor for RestClient.

Reply:
Agree, this constructor is migrated all the V1.  We need sort out of the build/serviceclient relationship.